src/mpi/errhan/errutil.c

```diff
@@ -254,10 +254,10 @@ int MPIR_Err_return_comm(MPIR_Comm * comm_ptr, const char fcname[], int errcode)
 
     if (errhandler == NULL) {
         /* Try to replace with the default handler, which is the one on
-         * MPI_COMM_WORLD.  This gives us correct behavior for the
-         * case where the error handler on MPI_COMM_WORLD has been changed. */
-        if (MPIR_Process.comm_world) {
-            comm_ptr = MPIR_Process.comm_world;
+         * MPI_COMM_SELF.  This gives us correct behavior for the
+         * case where the error handler on MPI_COMM_SELF has been changed. */
+        if (MPIR_Process.comm_self) {
+            comm_ptr = MPIR_Process.comm_self;
```

Comment:
Can we do it in a backward-compatible way? For example, if user have set a error handler on `MPI_COMM_SELF`, then use `comm_self`, other wise, use `comm_world` (especially if user have set a error handler on `MPI_COMM_WORLD`)